### PR TITLE
Switch to simpler looking loading spinner

### PR DESCRIPTION
**Pull request by skjnldsv**
New spinner. 
See discussion in ref issue.

Not compatible with ie (sigh) so i've added a fallback to gif!
I also added a small black loading icon: `icon-loading-small-dark`

ref #19091 
@oparoz @Henni @jancborchardt @PVince81

### CODE REVIEW

**Comment by ChristophWurst** [comment]
On Firefox, the spinner does not spin.

**Comment by skjnldsv** [comment]
Dammit! It works with direct image loading, but not as background-image... :/

**Comment by skjnldsv** [comment]
Okay, I switched from css animation to smil, it seems to be accepted as well as css animations:
http://caniuse.com/svg-smil

We still have the gif fallback for ie!

**Comment by oparoz** [comment]
Can't use SMIL

>As of Chrome 45 SMIL is deprecated and usage will result in a warning in the console. Support is expected to be dropped in some future version.

**Comment by skjnldsv** [comment]
I'm gonna stop coding now, and raise goats somewhere in the alps...

**Comment by oparoz** [comment]
>I'm gonna stop coding now, and raise goats somewhere in the alps...

Not everybody likes goat milk, you're going to need to include a hamster milk fallback

**Comment by mmattel** [comment]
> I'm gonna stop coding now, and raise goats somewhere in the alps...

This is a problem. There is no more free space left in the alps to raise goats. Too many frustrated programmers occupied everything. But there are maybe free caves availabe for meditation :smile:

**Comment by skjnldsv** [comment]
Okay, after a day of thinking while milking my goats, we should go for smil.
Considering the support still present for the next upcoming chrome & ff versions, we should be good for a while. Maybe the css-svg bug in ff will be fixed when smil will be dropped! :)
can i use support tables for html5 css3 etc

Let's review @oparoz @Henni @jancborchardt @PVince81!
Procedure:
 - Install commit
 - Just use owncloud and search for an inconvenient loader. Mails, contacts, user menu, files, gallery...

**Comment by skjnldsv** [comment]
For future reference:
This is also another way of using after:
```
.icon-loading::after {
	position: absolute;
	content: url('../img/loading-css.svg');
	width: 100%;
	height: 100%;
	top:50%;
	left: 50%;
	margin: -25px 0 0 -25px;
}
```

**Comment by MorrisJobke** [comment]
> Just use owncloud and search for an inconvenient loader. Mails, contacts, user menu, files, gallery...

Add `sleep(5);` to `apps/dav/appinfo/v1/webdav.php` and load the web UI ;)

**Comment by MorrisJobke** [comment]
> Add sleep(5); to apps/dav/appinfo/v1/webdav.php and load the web UI ;)

Also the while waiting for the search results to come in seems to be an old one. Beside that it looks awesome :)

**Comment by skjnldsv** [comment]
@MorrisJobke her eyou go! :D

**Comment by MorrisJobke** [comment]
Tested and works so far. I would merge this in and then fix the remaining stuff afterwards.

👍

**Comment by ChristophWurst** [comment]
@skjnldsv the spinner still does not spin on firefox. Also when switching folder in the file app the old spinner is shown

**Comment by ChristophWurst** [comment]
same result in IE11, the spinner is not animated

**Comment by ChristophWurst** [comment]
@MorrisJobke which browser did you test this with?

**Comment by MorrisJobke** [comment]
> @skjnldsv the spinner still does not spin on firefox. Also when switching folder in the file app the old spinner is shown

Oh ... damn. I also just noticed this in firefox.

👎 Then :(

**Comment by skjnldsv** [comment]
What? It was working this morning, what did I screwed up again! :(
I will take a look when back from work! Sorry!

**Comment by jancborchardt** [comment]
Very nice! :)

Only 3 additional things from my side:
- [ ] the spinning element seems to be thicker than the full circle in the dark version of the spinner. They should both have the same thickness.
- [ ] the spinner when loading Files (or going down into a folder) is still the old one.
- [ ] the spinner in the Contacts app, user management, and admin settings »Security & setup warnings« is HUGE, should only be as big as in Mail, Activity or Gallery for example. (Set that as the default size – there should be only that and then the small one.)

**Comment by skjnldsv** [comment]
- The thickness difference is done on purpose! Was more good looking this way I found. :)
- Ok, gonna change later
- It's because of the two different classes I talked to you about. The icon-xxx classes have a max height/width of 32px, but the .loading class does not have a dimension set.
- It's possible! :)

**Comment by jancborchardt** [comment]
> The thickness difference is done on purpose! Was more good looking this way I found. :)

It sticks a bit out though and looks more complex. A spinner should be as inconspicuous as possible. ;)

> It's because of the two different classes I talked to you about. The icon-xxx classes have a max height/width of 32px, but the .loading class does not have a dimension set.

Yeah, setting a max of 32*32 px is probably best for the loading class.

Thx about changing the other things! :)

**Comment by skjnldsv** [comment]
@jancborchardt Fixed all of it except the page change, can't do it from work, the proxy here prevent me to access my owncloud :'(

File: apps/user_ldap/css/settings.css [hunk 1]
```diff
@@ -179,12 +179,17 @@ select[multiple=multiple] + button {
...
 	color: orange;
 	font-weight: bold;
 }
 
-#ldap .ldap_saving img { height: 15px; }
+#ldap .ldap_saving .wizSpinner {
+	height: 16px;
+    display: inline-block;
+    position: absolute;
+    padding: 8px 0px;
```
Comment (jancborchardt) [inline apps/user_ldap/css/settings.css]: The indenting in these 3 lines seems to be different from the first one (tabs, not spaces).

Comment (jancborchardt) [inline apps/user_ldap/css/settings.css]: Also, when the value is 0, you do not need to specify a unit (like px). :)

File: core/search/css/results.css [hunk 2]
```diff
@@ -41,10 +41,8 @@
 }
 
 #searchresults .status .spinner {
-	height: 16px;
-	width: 16px;
-	vertical-align: middle;
-	margin-left: 10px;
+	float: left;
+    padding: 3px 10px;
```
Comment (jancborchardt) [inline core/search/css/results.css]: Here too, last line indent is wrong.

**Comment by jancborchardt** [comment]
Very nice! :) Except the indentation stuff and the spinner in Files, everything looks much better now!

File: apps/user_ldap/css/settings.css (hunk 1 above)
Comment (skjnldsv) [inline apps/user_ldap/css/settings.css]: Indeed indeed!

**Comment by skjnldsv** [comment]
@jancborchardt Better now? Is it spinning on ff?

**Comment by ChristophWurst** [comment]
Works, but only randomly on FF. When the image is not cached, it does not spin for some reason, which looks odd. :-/

**Comment by skjnldsv** [comment]
Should we assume the user will have loaded the svg at first and that it will be spinning on the rest of his browsing?

**Comment by skjnldsv** [comment]
@jancborchardt Files fixed.

**Comment by jancborchardt** [comment]
Nice! :+1: now only squash and this is good to merge

**Comment by oparoz** [comment]
Doesn't work on Firefox :-1:

**Comment by oparoz** [comment]
Actually, that's not true. The spinner just freezes sometimes, but that's normal FF behaviour.

I have one concern though, the spinner itself doesn't really work that well in a lot of situations, especially when the 16px version is used. It's a bit too discreet.

**Comment by jancborchardt** [comment]
So for the 16px version we might need a dedicated one – the line thickness should be the same as for the big spinner (3px?), not scaled down.

The Firefox issues I would say we should fix going forward, otherwise this will sit for too long. :\

**Comment by oparoz** [comment]
>So for the 16px version we might need a dedicated one – the line thickness should be the same as for the big spinner (3px?), not scaled down.

Sounds sane. I'm still worried about the big one as well, it's really thin compared to what's out there. There may be cases, especially when use as an overlay where it might not work that well.

>The Firefox issues I would say we should fix going forward, otherwise this will sit for too long. :\

I think we should try pre-loading as you suggested. I was able to reproduce the problem after clearing the cache. It's quite obvious when you switch app after logging in.

**Comment by MorrisJobke** [comment]
Looks really nice and works now in Firefox too :)

**Comment by MorrisJobke** [comment]
👍

File: apps/files/js/filelist.js [hunk 3]
```diff
@@ -2507,6 +2485,7 @@
 				}
 			});
 			fileUploadStart.on('fileuploadadd', function(e, data) {
+				console.log('XXXXXXX');
```
Comment (PVince81) [inline apps/files/js/filelist.js]: remove?

File: core/css/fixes.css [hunk 4]
```diff
@@ -138,3 +138,28 @@ select {
 	line-height: 38px;
 }
 
+/* Svg loader fixes */
+.ie .icon-loading {
+	background-image: url('../img/loading.svg');
```
Comment (oparoz) [inline core/css/fixes.css]: Should be gif

Comment (skjnldsv) [inline core/css/fixes.css]: Indeed! ^^

File: apps/files/js/filelist.js (hunk 3 above)
Comment (skjnldsv) [inline apps/files/js/filelist.js]: Yep, don't know how this got there! :O

**Comment by skjnldsv** [comment]
So, should we preload for ff?

**Comment by jancborchardt** [comment]
If that fixes the last issue, go for it! :)

**Comment by skjnldsv** [comment]
I will try, and we'll see :)

**Comment by skjnldsv** [comment]
Closed in favor of #24491

**Pull request closed without merging**